### PR TITLE
chore: introduce dev branch and pipelines

### PR DESCRIPTION
**Pull request by QuakeEye**
## Description

This pull request introduces the pipelines required for the proper functionality of the dev branch.

## Testability

Refer to this repo to show that it works:
https://github.com/QuakeEye/aplib.net-semver-dev/pull/12

main - https://github.com/QuakeEye/aplib.net-semver-dev

dev - https://github.com/QuakeEye/aplib.net-semver-dev/tree/dev

releases - https://github.com/QuakeEye/aplib.net-semver-dev/releases

## Checklist
- [x] Set the proper pull request name
- [x] Merged main into your branch

### CODE REVIEW

File: .github/workflows/pr-dev-to-main.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Create Pull Request from dev to main
+
+on:
+  workflow_dispatch:
+
```
Comment (SilasPeters) [inline .github/workflows/pr-dev-to-main.yml:5]: You can run this for every branch ever?

File: .github/workflows/pr-dev-to-main.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+jobs:
+  merge:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```
Comment (SilasPeters) [inline .github/workflows/pr-dev-to-main.yml]: I don't understand which branch is checked out. Main?

File: .github/workflows/pr-dev-to-main.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        id: previoustag
+        uses: WyriHaximus/github-action-get-previous-tag@v1.4.0
+        with:
+          fallback: v1.0.0
+
+      - name: Create Dev to Main Pull Request
+        run: |
+          git checkout dev
+          git pull
+          git checkout -b dev-${{ steps.previoustag.outputs.tag }}
```
Comment (SilasPeters) [inline .github/workflows/pr-dev-to-main.yml]: use `git switch -c`. Git switch is a replacement of checkout

**Review by SilasPeters — CHANGES_REQUESTED** [review]
Half review, have to go

File: .github/workflows/pr-dev-to-main.yml (hunk 1 above)
Comment (QuakeEye) [reply .github/workflows/pr-dev-to-main.yml:5]: That doesn't matter, it uses the correct branches on its own, it just needs to run.

File: .github/workflows/pr-dev-to-main.yml (hunk 2 above)
Comment (QuakeEye) [reply .github/workflows/pr-dev-to-main.yml]: Wherever this is ran, which is the default branch by the manual trigger, but it doesn't really matter, as it uses the correct branches on its own.

File: .github/workflows/pr-dev-to-main.yml (hunk 3 above)
Comment (QuakeEye) [reply .github/workflows/pr-dev-to-main.yml]: Alright

**Review by SilasPeters — APPROVED** [review]

**Review by joachimdekker — APPROVED** [review]
LGTM

**Pull request merged**